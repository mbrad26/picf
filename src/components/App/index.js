import React from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import LandingPage from '../Landing';
import * as ROUTES from '../../constants/routes';
import Navigation from '../Navigation';

const App = () => {
  return (
      <Router>
        <div>
          <Navigation />

          <hr/>

          <Switch>
            <Route exact path={ROUTES.LANDING} component={LandingPage} />
          </Switch>
        </div>
      </Router>
  )
}

export default App;
