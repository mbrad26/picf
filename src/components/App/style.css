body {
  font-family: 'Space Mono', monospace;
  margin: 0px;
}