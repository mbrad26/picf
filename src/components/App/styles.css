body {
  font-family: 'Space Mono', monospace;
  margin: 0px;
}

.error {
  color: #da0231;
}