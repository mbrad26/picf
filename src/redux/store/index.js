import { createLogger } from 'redux-logger';
import createSagaMiddleware from 'redux-saga';
import { createStore, applyMiddleware } from  'redux';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

const logger = createLogger();
const saga = createSagaMiddleware();

const store = createStore(
  rootReducer,
  {},
  applyMiddleware(saga, logger)
);

saga.run(rootSaga);

export default store;