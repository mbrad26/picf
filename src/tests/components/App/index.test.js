import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import store from '../../../redux/store/index';
import App from '../../../components/App';

describe('App', () => {
  it('renders snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>
      );

    expect(container.firstChild).toMatchSnapshot();
  });
});