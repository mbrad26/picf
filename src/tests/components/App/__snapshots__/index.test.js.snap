// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders snapshot 1`] = `
<div>
  <nav
    class="navbar navbar-expand navbar-light"
  >
    <span
      class="navbar-brand"
    >
      <a
        class="nav-link"
        href="/"
      >
        Picturesque
      </a>
    </span>
    <div
      class="ml-auto navbar-nav"
    >
      <a
        class="nav-link"
        href="/signup"
      >
        Sign Up
      </a>
      <a
        class="nav-link"
        href="/signin"
      >
        Sign In
      </a>
    </div>
  </nav>
  <div>
    <h1>
      Landing Page
    </h1>
  </div>
</div>
`;
